_posts/2018-10-23-artsy-auth.md

```diff
@@ -0,0 +1,269 @@
+---
+title: Everything You Ever Wanted To Know About Authentication at Artsy (But Didn't Know How To Ask)
+date: 2018-10-23
+categories: [tech, oauth, security, infosec, authentication]
+author: myk
+---
+
+# Everything You Ever Wanted To Know About Authentication At Artsy (But Didn't Know How To Ask)
+
+Hi! If you're reading this you're probably trying to figure out user authentication in an artsy application.
+Depending on what you're trying to do, you may need to set up [OAuth](https://oauth.net/2/) to allow users to log
+into your application by delegating authentication to gravity. Alternately, you may be working on an application
+that will only be called by other applications - in this case, you don't care about user authentication - you just
+need to establish permissions between services. In both cases you'll be working with
+[JSON Web Tokens (JWTs)](https://jwt.io/introduction/), and the difference is how the token you're looking at is
+generated.
+
+During user authentication, on login the user receives a token which explicitly grants their user account
+permission to use a specific application. For app authentication you'll create the token in advance and simply
+share it with whatever application you want to grant access to.
+
+In this document we'll first develop an understanding of what OAuth is and how it works. Then we'll examine the
+tokens we're using to get a better sense of what kind of information we have to work with. Finally, we'll go into
+how to set up authentication for users and for applications, building on the knowledge we've established.
+
+<!-- more -->
+
+### OAuth
+
+So you've got some publicly facing web application that you want to protect - maybe only Artsy users should have
+access to this service, or part of it, or something like that. Maybe you only want Artsy employees to have access.
+Maybe you only want to allow requests from other artsy applications. Regardless, you need to figure out how to
+provision access to the functionality you've deployed. At Artsy, we use OAuth for that.
+
+OAuth is a paradigm that allows one service to delegate authentication to another one. You've probably used this
+before, even if you don't realize it. Whenever you use your Twitter account to log in to Medium, or your Facebook
+account to log in to Tinder, or whatever, you're relying on delegated authentication. Medium doesn't have to know
+much about who you are as long as they can trust Twitter to tell them, etc.
```

Comment:
If you have any recommended resources for OAuth, or the OAuth implementations at the companies mentioned here, maybe we could link to them?